From 2cde955c1a6630c23c6133ed5ff166d82f65ca15 Mon Sep 17 00:00:00 2001
From: Filip Kotoucek <filip.kotoucek@prusa3d.cz>
Date: Fri, 20 May 2022 20:05:02 +0200
Subject: [PATCH] dts: remap wifi pins

- define dldo4 voltages
- WL_HOST_WAKE on PD15
- WL_REG_ON on PD16

---
 .../boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts      | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts
index ba533614188f..25a4d6411a94 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts
@@ -116,7 +116,7 @@ simple-audio-card,codec {
 
 	wifi_pwrseq: wifi_pwrseq {
 		compatible = "mmc-pwrseq-simple";
-		reset-gpios = <&pio 7 11 GPIO_ACTIVE_LOW>; /* PH11 */
+		reset-gpios = <&pio 3 16 GPIO_ACTIVE_LOW>; /* PD16 */
 	};
 };
 
@@ -376,7 +376,7 @@ brcmf: wifi@1 {
 		reg = <1>;
 		compatible = "brcm,bcm4329-fmac";
 		interrupt-parent = <&pio>;
-		interrupts = <3 6 IRQ_TYPE_LEVEL_LOW>; /* PD6 */
+		interrupts = <3 15 IRQ_TYPE_LEVEL_LOW>; /* PD15 */
 		interrupt-names = "host-wake";
 	};
 };
@@ -487,6 +487,12 @@ &reg_dldo3 {
 	regulator-name = "dsi0-3.3v-ctp";
 };
 
+&reg_dldo4 {
+	regulator-min-microvolt = <3300000>;
+	regulator-max-microvolt = <3300000>;
+	regulator-name = "vcc-wifi";
+};
+
 &reg_drivevbus {
 	regulator-name = "usb0-vbus";
 	status = "okay";
